src/libraries/Delegator.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+
+
+contract Delegator is DelegationTarget {
+
+    function Delegator(address controllerAddress, bytes32 keyVal) {
+        controller = Controller(controllerAddress);
+        controllerLookupName = keyVal;
+    }
+
+    // We currently only support a single non-array return value
+    // If we truly need to return more advanced sets of data we could follow
+    // the pattern demonstrated here: https://gist.github.com/Arachnid/4ca9da48d51e23e5cfe0f0e14dd6318f
+    function() payable {
+        // Do nothing if we haven't properly set up the delegator to delegate calls
+        if (controllerLookupName == 0) {
+            return;
+        }
+
+        // Get the delegation target contract
+        address target = controller.lookup(controllerLookupName);
+
+        assembly {
+            let o_code := mload(0x40) //Memory end
```

Comment:
We should apply the same code style in inline assembly as the rest, which means `_` prefix parameters and locals.